scripts/linters/pre_commit_linter.py

```diff
@@ -508,32 +532,59 @@ def main(args=None):
     # third party libraries have their own ways to lint at their fastest
     # (ie. might parallelize on their own)
 
+    error_messages = []
+
     # Concurrency limit: 25.
-    concurrent_task_utils.execute_tasks(tasks_custom, custom_semaphore)
-    # Concurrency limit: 2.
-    concurrent_task_utils.execute_tasks(
-        tasks_third_party, third_party_semaphore)
+    custom_task_execution_failed = False
+    try:
```

Comment:
Why we need a try-expect block here? I tried looking into the `concurrent_task_utils.py` and I found we already handle exceptions while running tread 
![image](https://user-images.githubusercontent.com/16653571/81841605-572e2200-9568-11ea-95a2-0b97a0cf6240.png)

is this what we are doing here^? If yes, why don't we use concurrent_task_utils.ALL_ERRORS?

Reply:
This try-except block is for `task_execute` method. If there is any error during execution of this function it will be caught by this try-except block.

We can't use ALL_ERRORS because it does not give us full stack trace of the errors. It only return what has happend i.e. errors like `index out of range`

Reply:
> I used try-except block for task_execute method not for TaskThread

By `task_execute` you mean `concurrent_task_utils.execute_tasks`, right? if yes, then `execute_tasks` is internally ruuning tasks (made of TaskThread class).

> If there is an error during execution of this function it will be caught by this try-except block.

We aren't expecting to have an error in the `execute_tasks` method (though there can be inside the `task.start` but we have already handled it, right?)

> We can't use ALL_ERRORS because it does not give us full stack trace of the errors. ALL_ERRORS only return what has happend i.e. errors like index out of range not full stack trace.

In that case, can we make changes such that ALL_ERRORS reports all the errors? [I think this will be a more general approach and we don't have to add  try-except block everywhere!]

Reply:
Removed try-except from here.